File: lib/py/uapi/internal/binary/BinaryEncoding.py
from typing import Dict


class BinaryEncoding:
    def __init__(self, binary_encoding_map: Dict[str, int], checksum: int) -> None:
        self.encode_map: Dict[str, int] = binary_encoding_map
        self.decode_map: Dict[int, str] = {
            v: k for k, v in binary_encoding_map.items()}
        self.checksum: int = checksum

File: lib/ts/uapi/internal/binary/BinaryEncoding.ts
export class BinaryEncoding {
    encodeMap: Record<string, number>;
    decodeMap: Record<number, string>;
    checksum: number;

    constructor(binaryEncodingMap: Record<string, number>, checksum: number) {
        this.encodeMap = binaryEncodingMap;
        this.decodeMap = {};
        for (const [key, value] of Object.entries(binaryEncodingMap)) {
            this.decodeMap[value] = key;
        }
        this.checksum = checksum;
    }
}
